### PR TITLE
Update ledger-autosync to 1.0.3

### PR DESCRIPTION
**Pull request by DamienCassou**
###### Motivation for this change

Update ledger-autosync

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by DamienCassou** [comment]
Result of `nixpkgs-review pr 128203` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>ledger-autosync</li>
    <li>python38Packages.ofxclient</li>
    <li>python38Packages.ofxparse</li>
    <li>python39Packages.ofxclient</li>
    <li>python39Packages.ofxparse</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 128203` at 0218d454 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built successfully:</summary>
  <ul>
    <li>ledger-autosync</li>
    <li>python38Packages.ofxclient</li>
    <li>python38Packages.ofxparse</li>
    <li>python39Packages.ofxclient</li>
    <li>python39Packages.ofxparse</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/ofxparse/default.nix:20:3:
```
   |
20 |   meta = with lib; {
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/applications/office/ledger-autosync/default.nix [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
 
-  patches = [
-    # ledger-autosync specifies an URL for its ofxparse
-    # dependency. This patch removes the URL to only use the
-    # `ofxparse` name. This works because nixpkgs' version of ofxparse
-    # is more recent than the latest release.
-    ./fix-ofxparse-dependency.patch
+  nativeBuildInputs = with python3Packages; [
+    poetry-core
+    pytest
```
Comment (SuperSandro2000) [inline pkgs/applications/office/ledger-autosync/default.nix:25]:
```suggestion
```

File: pkgs/applications/office/ledger-autosync/default.nix [hunk 2]
```diff
@@ -46,8 +51,11 @@ python3Packages.buildPythonApplication rec {
   # Checks require ledger as a python package,
   # ledger does not support python3 while ledger-autosync requires it.
   checkInputs = with python3Packages; [ ledger hledger nose mock ];
```
Comment (SuperSandro2000) [inline pkgs/applications/office/ledger-autosync/default.nix:53]:
```suggestion
  checkInputs = with python3Packages; [ ledger hledger nose mock pytestCheckHook ];
```

File: pkgs/applications/office/ledger-autosync/default.nix [hunk 3]
```diff
@@ -46,8 +51,11 @@ python3Packages.buildPythonApplication rec {
...
+
+  # Disable some non-passing tests:
+  # https://github.com/egh/ledger-autosync/issues/127
   checkPhase = ''
-    nosetests -a generic -a ledger -a hledger
+    pytest -k 'not test_payee_match and not test_args_only'
   '';
```
Comment (SuperSandro2000) [inline pkgs/applications/office/ledger-autosync/default.nix:59]: Use pytestCheckHook instead.

Comment (DamienCassou) [reply pkgs/applications/office/ledger-autosync/default.nix]: Much nicer, thank you for your suggestion.

**Comment by DamienCassou** [comment]
Thank you very much for your suggestions @SuperSandro2000. I took all of them into account.

**Review by SuperSandro2000 — APPROVED** [review]
diff LGTM

**Pull request merged**